src/UI/UIView.cpp

```diff
@@ -0,0 +1,520 @@
+
+     ImGui::Checkbox("Render Volume data", &m_renderVolume);
+
+     int numVolumes = m_controller_app.get_num_volumes();
+
+    
+
+    if (numVolumes > 0)
+    {
+      if (ImGui::SmallButton("New")) {
+        tfn_widget.push_back(TransferFunctionWidget());
+        tfn_widget_multi.push_back(TransferFunctionMultiChannelWidget());
+        int index = m_selectedTrFn.size();
+        m_selectedTrFn.push_back(std::vector<bool>(numVolumes));
+        for (int i = 0; i < numVolumes; i++)
+        {
+          m_selectedTrFn[index][i] = false;
+        }
+      };
+      ImGui::SameLine();
+      if (ImGui::SmallButton("Remove")) {
+
+
+      };
+
+      ImGui::BeginTable("##Transfer Function Editor", 3);
+      ImGui::TableSetupColumn("Name");
+      for (int column = 0; column < numVolumes; column++)
+      {
+        ImGui::TableSetupColumn(m_dataLabels[column].c_str());
+      }
+      ImGui::TableHeadersRow();
+
+      for (int row = 0; row < tfn_widget.size(); row++)
+      {
+        ImGui::TableNextRow();
+        for (int col = 0; col < numVolumes + 1; col++)
+        {
+          ImGui::TableSetColumnIndex(col);
+          if (col == 0)
+          {
+            char buf[32];
+            sprintf(buf, "TF%d", row);
+            if (ImGui::SmallButton(buf)) {
+              std::cout << buf << std::endl;
+              m_selectedTrnFnc = row;
+            };
+
+          }
+          else
+          {
+            char buf[32];
+            sprintf(buf, "##On%d%d", col, row);
+            bool b = m_selectedTrFn[row][col - 1];
+            ImGui::Checkbox(buf, &b);
+            m_selectedTrFn[row][col - 1] = b;
+            if (b)
+            {
+              std::cout << "ddd" << std::endl;
+            }
+          }
+
+        }
+      }
+
+      ImGui::EndTable();
+
+      ImGui::Checkbox("use transferfunction", &m_use_transferfunction);
+      if (m_use_transferfunction) {
+        bool is_multi_channel = m_controller_app.data_is_multi_channel();
+        if (is_multi_channel)
+        {
+          for (int i = 0; i < 3; i++) {
+            if (m_animated)
+            {
+              /*	
+                frame by frame animation
+
+                unsigned int active_volume = floor(m_frame);
+                unsigned int active_volume2 = ceil(m_frame);
+                double alpha = m_frame - active_volume;
+                if (active_volume < m_volumes[m_selectedVolume].size() && active_volume2 < m_volumes[m_selectedVolume].size())
+                {
+                  tfn_widget_multi[m_selectedVolume].setBlendedHistogram(
+                    m_volumes[m_selectedVolume][active_volume]->getTransferfunction(i),
+                    m_volumes[m_selectedVolume][active_volume2]->getTransferfunction(i), alpha, i);
+                }*/
+
+            }
+            else {
+              /*		tfn_widget_multi[m_selectedVolume].setHistogram(m_volumes[m_selectedVolume][0]->getTransferfunction(i), i);*/
+            }
```

Comment:
Do you need this if statement still?

Reply:
Let's keep it for now. I'll remove it in the next PR. There are things I want to test without the need for rolling back specific code.